src/Donations/Repositories/DonationMetaRepository.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Give\Donations\Repositories;
+
+use Give\Framework\Database\DB;
+use Give\Framework\QueryBuilder\QueryBuilder;
+use Give\Framework\Support\Facades\Str;
+
+/**
+ * @unreleased
+ */
+class DonationMetaRepository
+{
+    /**
+     * @unreleased
+     * @return mixed|null
+     */
+    public function get(int $donationId, string $metaKey)
+    {
+        $query = $this->prepareQuery()
+            ->where("donation_id", $donationId)
+            ->where("meta_key", $metaKey)
+            ->get();
+
+        if (!$query) {
+            return null;
+        }
+
+        $value = $query->meta_value;
+
+        if (Str::isJson($value)) {
+            return json_decode($value, false);
+        }
+
+        return $value;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function upsert(int $donationId, string $metaKey, $metaValue): void
+    {
+        if (is_array($metaValue) || is_object($metaValue)) {
+            $metaValue = json_encode($metaValue);
+        }
+
+        $exists = $this->exists($donationId, $metaKey);
```

Comment:
Suggestion from @alaca:

Explore using this for one query: https://dev.mysql.com/doc/refman/5.7/en/insert-on-duplicate.html

Reply:
@jon Instead of adding the `upsert` method to `DonationMetaRepository`, we can add that to our QB. See stellar implementation [stellarwp/db](https://github.com/stellarwp/db/blob/main/src/DB/QueryBuilder/Concerns/CRUD.php#L62)